### ISSUE TITLE
Suggestion to reduce number of clicks

### ISSUE THREAD

**Issue by GitBoudewijn** [dsohler/oneclick-downloader#3]
Hey,

I have a suggestion to reduce the number of clicks required to download images.

It would be nice if when you open the extension it would begin with the current tab (the one you were looking at when you clicked the button) already selected, since this is almost always the tab you want to download from. Also in the second step I would like it to begin with everything selected as opposed to nothing selected, but this is maybe more of a personal preference.

**Comment by dsohler** [comment]
Actually I was thinking about both of the suggestions you made during initial creation of the extension. I think of this thing more as app than an extension but since apps have a LOT less permissions in the browser it simply has to be an extension.

On the page where you select the links for downloading there is a button for selecting all of the links for downloading.

Maybe I’ll find another solution for both of the steps in one of the future releases.

**Comment by dsohler** [comment]
In 4554d10 I implemented your suggestion of pre-checking the currently active tab.

Personally I think I’ll leave the second step as it is because there already is a way to check all of the links at once. One might not always want to download all images and in the end we’re talking about one single click here :)

I’ll release this version to the store after fixing the definition issues.

**Comment by GitBoudewijn** [comment]
Thanks, it works!

You're right that one click isn't that much of a problem. I would like to eliminate the need to scroll though. Let's take an extreme example: Say you have 1000 tabs open and you want to download 1000 images. First you have to scroll down all the way in order to click the button, and then you have to scroll all the way down again to click the buttons to select all and download.

The solution to this is to either put the buttons on the top or to have the buttons fixed on the bottom and have the list of links in an element with a scrollbar. You could also make it more intuitive by putting the buttons that further the process (scrape, download, restart and close) to the right side, while having the other buttons that apply to the current step on the left. This way you will be able to predict the position of the buttons in the next step, so you reduce the need to process the screen with your eyes.

**Comment by dsohler** [comment]
During the day I was thinking about a setting in the options dialog allowing users to set their desired behavior. Another thought was using some kind of [toggle mechanism](http://codepen.io/Linuxdirk/pen/XjoxvZ?editors=0010) even if it would break with the overall look and feel a little.

If you have 1000 tabs open each with 1000 images you might run into other issues :) Moving the buttons to any side of the window and fix them there is quite easy. Maybe I go for it in a later version when I overhaul the general appearance.

**Comment by GitBoudewijn** [comment]
More settings is always good. Maybe you could even have a setting to skip the first step entirely (I've never downloaded from more than one tab at a time). Maybe it could have two modes, one that's meant to stay open (you show the first step and you hide the "close window" button) and another that's meant for one time use (you hide the first step and the "start over" button). But maybe that's a little too complicated.:P

Yeah that's true but I meant 1000 images in total, so you would have to scroll down 1000 items each time. Would be great if the buttons were in a fixed place so you wouldn't have to scroll. On some forums you can easily have a page with 300+ images.

Another thing that would be really helpful is if there was a count of the number of selected or completed items, either at the top or at the bottom.
For example:
**"1/4 tabs selected"**
**"20/50 links selected"**
**"19/20 images downloaded"**

**Comment by dsohler** [comment]
Added startup behavior controls with c05f778 (see this commit’s description for potential issues). The behavior options are the following ones:

1. **default** - The usual app-like behavior where nothing is pre-selected or automatically scraped
2. **current** - … where the currently active tab gets selected automatically.
3. **links** - The currently active tab gets selected and automatically scraped as if the user clicked the button for scraping it.

The **links** option basically disables the possibility to re-use the window for anything else then the current tab. All buttons that normally go back to the overview in some way are simply causing re-scraping of the links that are in the current tab.

This version will be pushed to the store as soon as the “next definition update” milestone is reached.

**Comment by GitBoudewijn** [comment]
Hey,

Great idea! But it seems that sometimes the setting doesn't stick. I haven't noticed a pattern yet but maybe you can try to reproduce the following:

1. Open Chrome
2. Go to **chrome://extensions**
3. Open the options menu, select the third option, save and close the option menu
4. Start the extension (it now tries to scan the page)
5. Close Chrome (I have the setting to keep Chrome running in the background turned off so maybe you have to kill the process as well)
6. Open Chrome
7. Go to **chrome://extensions**
8. Start the extension (it now asks you to select tabs)

**Comment by dsohler** [comment]
Yes, this is a quick workaround. Unfortunately there seems no other way except stuffing the whole extension into the scope of the configuration option function - which I totally do not want.

Does it work when reloading the already open extension window by pressing F5 or one of the buttons?
